python/meep.i

```diff
@@ -489,6 +490,24 @@ void _get_eigenmode(meep::fields *f, double omega_src, meep::direction d, const
     (std::complex<double> *arr, size_t dim1, size_t dim2, size_t dim3)
 };
 
+%typemap(in) meep::source_indicator *
+{ if ($input==Py_None)
+   $1=NULL;
+  else
+   { GEOMETRIC_OBJECT obj;
+     if(!py_gobj_to_gobj($input,&obj))
+      SWIG_fail;
+     $1=new object_source_indicator(obj);
+   }
+}
+
+%typecheck(SWIG_TYPECHECK_POINTER) meep::source_indicator * 
+{ $1 = 1; }
```

Comment:
Here we should probably check if `$input` is an instance of a `mp.GeometricObject`. We need a function `py_geometric_object` similar to [py_source_time_object](https://github.com/stevengj/meep/blob/master/python/meep.i#L90), and then this typemap can be

```c++
{
  $1 = PyObject_IsInstance($input, py_geometric_object()) || $input == Py_None;
}
```